lib/spack/spack/solver/asp.py

```diff
@@ -2510,7 +2541,12 @@ def _reusable_specs(self):
 
             # Specs from buildcaches
             try:
-                index = spack.binary_distribution.update_cache_and_get_specs()
+                packages = spack.config.get("packages")
+                index = [
+                    s
+                    for s in spack.binary_distribution.update_cache_and_get_specs()
+                    if not s.external or _is_reusable_external(packages, s)
+                ]
```

Comment:
This should have a comment explaining that making the dependency reusable/not effectively controls whether the dependent is reusable/not. It's too slick to go without a comment.